src/fromager/context.py

```diff
@@ -119,13 +119,40 @@ def add_to_build_order(
             # converted to JSON without help.
             json.dump(self._build_stack, f, indent=2, default=str)
         with open(self._constraints_filename, "w") as f:
-            for step in self._build_stack:
+            for step in self._process_constraints(self._build_stack):
                 comment = " ".join(
                     f"-{dep_type}-> {req.name}({version})"
                     for dep_type, req, version in step["why"]
                 )
                 f.write(f'{step["dist"]}=={step["version"]}  # {comment}\n')
```

Comment:
After processing the build_stack wouldn't we have to comment out any duplicates?

Reply:
I discussed this  with Doug, and he said `Maybe we don't pick for now, and just list the ones with duplicates at the top of the file so it's easy for a person to pick them`

Reply:
I tried sorted() but we want all the duplicates at the top of file followed by non duplicates and sorted doesn't do that even though it groups items

Reply:
Oh so you want any pkg with duplicates to be on the top of the file not just grouped together in the file?

Reply:
Yeah!

Reply:
You could try this:
```python
# count the number of occurrences of a given dist
count = collections.Counter([item["dist"] for item in self._build_stack])
# first group all same dist items together as it will help with handling cases when 2 dists have the same count
grouped_build_stack = sorted(input, key=lambda item: item["dist"])
# sort based on the count
sorted(grouped_build_stack, key=lambda item: count[item["dist"]], reverse=True)
```

Reply:
Okay, let me try this